src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MeshCypherDecoder.java

```diff
@@ -0,0 +1,58 @@
+package uk.nhs.digital.nhsconnect.nhais.mesh;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.nhs.digital.nhsconnect.nhais.model.mesh.MeshMessage;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.nhais.parse.EdifactParser;
+
+@Component
+public class MeshCypherDecoder {
```

Comment:
This shouldn't be needed. These properties can be added to the TranslatedInterchange class.